### ISSUE TITLE
Task Log Config Validation: Log usage calculation incorrect

### ISSUE THREAD

**Issue by jaltavilla** [hashicorp/nomad#18933]
I believe that the calculation of disk space needed by logs when comparing to available ephemeral disk is incorrect. Either that or the documentation on [max_files](https://developer.hashicorp.com/nomad/docs/job-specification/logs#max_files) and [ephemeral_disk](https://developer.hashicorp.com/nomad/docs/job-specification/ephemeral_disk) is incorrect. I think this calculation needs to be doubled: [https://github.com/hashicorp/nomad/blob/66fbc0f67e47b3fc5f6007e624173e18905f9b63/nomad/structs/structs.go#L7438](https://github.com/hashicorp/nomad/blob/66fbc0f67e47b3fc5f6007e624173e18905f9b63/nomad/structs/structs.go#L7438)

The max files documentation states that this value is per stream. So without the multiplier, it's only checking that there is room for one stream on disk. I checked where the max files are set to make sure files aren't doubled there instead. That didn't seem to be the case and constructing the TaskLogger objects doesn't halve the value. I could have still missed a detail somewhere.

I think this is pretty minor as the default settings are safe. Ephemeral disk defaults to 300mb which exceeds the default log config's 200mb (2 * 10 files * 10mb). 

I'm sorry I didn't make a pr for this, but our legal department hasn't approved of me signing the CLA yet. Thanks!

**Comment by lgfa29** [comment]
Hi @jaltavilla 👋 

Thanks for the report and for being willing to submit a PR, I know how the CLA can be a bit of burden on contributions 😓 

As for the issue, I think you're absolutely right. Nomad does end up creating two log streams and so, in theory, the disk space reserved for logs should be doubled.

The only thing that I'm not sure about is the impact of a change like this. Nomad doesn't have any hard limits on how much disk a task is using, so the value of `ephemeral_disk` is more of a pinky swear approximation of how much disk a task is expected to use that the Nomad scheduler uses to avoid overloading a client with disk-heavy allocations.

But I believe some folks don't have this problem (either they have separate disk management solutions, or their tasks don't make heavy use of disk),  and so they reduce their `ephemeral_disk` values to the minimum allowed to maximize alloc density. Doubling this value could then cause jobs like this to start failing

I will check with team for any extra thoughts and report back to you once I get an answer.

**Comment by lgfa29** [comment]
Hello again @jaltavilla 👋 

I discussed this with the rest of the team and we decided it may be best to leave these values as they are for now.

The main reason being what I mentioned in my previous comment (it may cause a significant impact on users without much benefit) but it's also an area where we have some possible improvements coming in the future where we may be able to address this properly, such as storage quotas and log plugins.

So I will leave this open until we have a better answer for it, but just to set the proper expectations we don't expect it to change for now.

But thank you again for the great find and reporting it to us!

**Comment by jaltavilla** [comment]
Thanks for looking into this! That makes sense to me. The only reason I even looked at the code was to make sure we understood what lowering ephemeral disk in our jobs would do. (We were experiencing blocked allocations due to disk space with the default settings and our autoscaling policies.)

**Comment by lgfa29** [comment]
> We were experiencing blocked allocations due to disk space with the default settings and our autoscaling policies.

Yeah, exactly. This can often be a surprising behaviour, and doing the right math would double the default requirements and cause this happen more often 😅
